src/main/scala/org/constellation/checkpoint/CheckpointAcceptanceService.scala

```diff
@@ -199,8 +207,11 @@ class CheckpointAcceptanceService[F[_]: Concurrent: Timer](
         allowedToAccept <- awaitingBlocks.toList.filterA { c =>
           AwaitingCheckpointBlock.areParentsSOEAccepted(checkpointParentService.soeService)(c.checkpointBlock)
         }.flatMap(_.filterA { c =>
-          AwaitingCheckpointBlock.areReferencesAccepted(transactionService.transactionChainService)(c.checkpointBlock)
-        })
+            AwaitingCheckpointBlock.areReferencesAccepted(transactionService.transactionChainService)(c.checkpointBlock)
+          })
+          .map(_.filter { _ =>
+            blacklistedTxs.isEmpty
```

Comment:
Why are you checking `blacklistedTxs` for currently accepted `cb` in the list of awaiting checkpoint blocks?
It should be implemented in `AwaitingCheckpointBlock.hasNoBlacklistedTxs` and checked for each cb in this list